sdks/python/apache_beam/examples/snippets/snippets_test.py

```diff
@@ -696,12 +696,31 @@ def test_model_group_by_key(self):
     self.assertEqual([str(s) for s in expected], self.get_output(result_path))
 
   def test_model_co_group_by_key_tuple(self):
-    email_list = [['a', 'a@example.com'], ['b', 'b@example.com']]
-    phone_list = [['a', 'x4312'], ['b', 'x8452']]
+    # [START model_group_by_key_cogroupbykey_tuple_inputs]
+    email_list = [
+        ('amy', 'amy@example.com'),
+        ('carl', 'carl@example.com'),
+        ('julia', 'julia@example.com'),
+        ('carl', 'carl@email.com'),
+    ]
+    phone_list = [
+        ('amy', '111-222-3333'),
+        ('james', '222-333-4444'),
+        ('amy', '333-444-5555'),
+        ('carl', '444-555-6666'),
+    ]
+    # [END model_group_by_key_cogroupbykey_tuple_inputs]
     result_path = self.create_temp_file()
     snippets.model_co_group_by_key_tuple(email_list, phone_list, result_path)
-    expect = ['a; a@example.com; x4312', 'b; b@example.com; x8452']
-    self.assertEqual(expect, self.get_output(result_path))
+    # [START model_group_by_key_cogroupbykey_tuple_outputs]
+    contact_lines = [
+        "amy; ['amy@example.com']; ['111-222-3333', '333-444-5555']",
+        "carl; ['carl@example.com', 'carl@email.com']; ['444-555-6666']",
+        "james; []; ['222-333-4444']",
+        "julia; ['julia@example.com']; []",
+    ]
+    # [END model_group_by_key_cogroupbykey_tuple_outputs]
+    self.assertEqual(contact_lines, self.get_output(result_path))
```

Comment:
This assert may fail. There is no ordering guarantee in the resulting lists. Unless you sort them in your lambda, these asserts will depend on the runner to sort. And even in direct runner may occasionally flake because of threading.

Reply:
Should I change it to use `apache_beam.testing.util.assert_that` and `apache_beam.testing.util.equal_to`? For correctness, I can also update the rest of the asserts in this file since they are all using `self.assertEqual` too.

Reply:
There are two options:
- Use assert_that with equal_to. equal_to will do the sorting. problem with that is, assert_that needs to be part of the pipeline. You cannot assert on the output.
- In the pipeline, where you convert to string, sort the results.

I have not looked at other examples. If they have only 0 or 1 result, they are fine. If they assert of more than 1 result, without sorting somewhere, we need to update them as well.

Reply:
No it is not deterministic. `get_output` will sort higher level items correctly `"amy; ['amy@example.com']; ['111-222-3333', '333-444-5555']"` and `"carl; ['carl@example.com', 'carl@email.com']; ['444-555-6666']"` for example. But it will not sort within the item. It is possible that latter item become `"carl; ['carl@email.com', 'carl@example.com']; ['444-555-6666']"`

Reply:
I see, I'll just write the sorted emails and phones to the file.